### ISSUE TITLE
Rule Change: Add suggestions to `no-new-wrappers` and `no-new-native-nonconstructor`

### ISSUE THREAD

**Issue by Pixel998** [eslint/eslint#21040]
### What rule do you want to change?

`no-new-wrappers`, `no-new-native-nonconstructor`

### What change do you want to make?

Implement suggestions

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
/* eslint no-new-wrappers: "error", no-new-native-nonconstructor: "error" */

const str = new String("hello");
const sym = new Symbol("foo");
```

### What does the rule currently do for this code?

Both rules report errors for using `new`, but neither provides suggestions to fix the code.

### What will the rule do after it's changed?

The rules will provide suggestions to remove the `new` keyword.

### Participation

- [x] I am willing to submit a pull request to implement this change.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

**Comment by mdjermanovic** [comment]
> The rules will provide suggestions to remove the `new` keyword.

Makes sense to me for `no-new-native-nonconstructor`, as code like `foo = new Symbol()` is certainly invalid since it throws an error, and the intent was most likely to do `foo = Symbol()`.

I'm not sure about `no-new-wrappers` because removing `new` makes an assumption that the code was intended to just convert a value to another primitive type, which isn't the case with code such as `foo = new String()` or `foo = new String("Hello world")`. Also, in code such as `foo = new String(bar)`, we don't know if `bar` is already a string value and thus the intent was indeed to create a wrapper, in which case removing `new` doesn't make sense.

@eslint/eslint-team thoughts?

**Comment by ivangray271-lgtm** [comment]
I agree with your assessment regarding no-new-native-nonconstructor since
new Symbol() is always invalid.

For no-new-wrappers, I understand the concern about assuming the user's
intent. Perhaps we could limit the suggestions to cases where the intent is
more likely for type conversion, or keep it as an optional suggestion
rather than an automatic fix.

What does the rest of the @eslint/eslint-team (
https://github.com/orgs/eslint/teams/eslint-team) think?

**Comment by DMartens** [comment]
I agree with both rules having a suggestion fixer.
I think `no-new-wrappers` only should suggest a different fix based on whether the type statically matches:
- Matches: `new String("string")` -> `"string"` (remove call all together)
- maybe matches / does not match: `new String(bar)` -> `String(bar)` (convert to a cast)

**Comment by mdjermanovic** [comment]
> * Matches: `new String("string")` -> `"string"` (remove call all together)

This case certainly looks like an intent to create a wrapper object, and in existing code, changing it to a string value would probably break the code.

> * maybe matches / does not match: `new String(bar)` -> `String(bar)` (convert to a cast)

This also could be intent to create a wrapper object rather than an attempt to just convert the value.

Overall, while this would be a suggestion and therefore not necessarily safe, I think it's usually more likely that some other change to the code is needed than just removing the `new` keyword.
